source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1215,6 +1220,12 @@ FilterHeadersStatus Filter::encodeHeaders(ResponseHeaderMap& headers, bool end_s
   // local reply.
   mergePerRouteConfig();
 
+  // If there is no external processing configured in the encoding path,
+  // closing the gRPC stream if it is still open.
+  if (encoding_state_.noExternalProcess()) {
+    closeStreamMaybeGraceful();
```

Comment:
if `noExternalProcess`  can you simply return FilterHeadersStatus::Continue here and avoids the rest of handling in this function?